### PR TITLE
#160580553 users can subscribe to read all articles

### PR DESCRIPTION
**Pull request by emekafredy**
# What does this PR do?
- Create endpoints for users to subscribe on the platform

# Description of Task to be completed?
- Create Subscription model and its association with users
- Integrate payment with stripe
- Update get-article endpoint, so only users with a valid subscription plan can read more than five articles
- Create endpoints:
   - Basic Subscription ``` POST api/subscription/subscribe/<basic>```
   - Premium Subscription ``` POST api/subscription/subscribe/<premium>```
- Write unit tests for feature
- Update swagger doc

# How should this be manually tested?
- Clone the repo
- cd into the 'Thor-ah'
- Run 'npm install'
- Use the .env.sample format to create a .env file
- Run 'npm run migrate'
- Run 'npm run start:dev'

### CODE REVIEW

**Comment by emekafredy** [comment]
> `Your subscription is not active, Please renew to get more articles`
> I think this works better for a response that we'll be sending to the frontend.

Yeah. That's nice

File: server/tests/controllers/users.js [hunk 1]
```diff
@@ -46,6 +46,8 @@ const invalidPassword = {
   email: 'emeka1@gmail.com',
   password: 'sdfghgfdkjshfbjbsfjbdmnbdkjfgeugfuefguefge'
 };
+const worngEmail = { email: 'xyz@mail.com', password: 'passtes' };
```
Comment (d-beloved) [inline server/tests/controllers/users.js]: There is a typo here, you have ```worngEmail```, you might want to change it to ```wrongEmail``` to aid easy interpretation of your code, and please don't forget to change it at all the places it was referenced too

**Review by d-beloved — CHANGES_REQUESTED** [review]
A very nice work you have done, please make corrections to the typo error I pointed out.

File: server/tests/controllers/users.js (hunk 1 above)
Comment (emekafredy) [reply server/tests/controllers/users.js]: Thanks bro. Will correct this

**Comment by emekafredy** [comment]
> In line 11, you have the following as comment
> 
> ```
> /**
>  *
>  * @class Subscription
>  *
> */
> ```
> .....but the class I saw in line 13 is "SubscriptionClass"

Okay man. I will correct this

**Review by d-beloved — APPROVED** [review]

**Pull request merged**